src/isyntax/isyntax_reader.h

```diff
@@ -21,7 +22,9 @@ typedef struct isyntax_cache_t {
     int allocator_block_height;
 } isyntax_cache_t;
 
-uint32_t* isyntax_read_tile_bgra(isyntax_t* isyntax, isyntax_cache_t* cache, int scale, int tile_x, int tile_y);
+// TODO(avirodov): can this ever fail?
```

Comment:
Yes, I think so, e.g. if the decompression of the codeblocks fails (for example, if trying to decompress corrupt 
or garbage codeblock). I guess we should probably detect that and escalate that up the call chain?

Reply:
I think so too. I prefer to defer this problem to when we need to virtualize the filesystem access (see https://github.com/openslide/openslide/pull/437#discussion_r1174099526, maybe will be needed soon). Then we can figure out what errors the virtual FS can return, how to propogate them, and how to expose them here.

So for now I'll leave it as a todo.